circuits/cpp/src/aztec3/circuits/abis/global_variables.hpp

```diff
@@ -0,0 +1,91 @@
+#pragma once
+#include "function_data.hpp"
+#include "tx_context.hpp"
+
+#include "aztec3/utils/array.hpp"
+#include "aztec3/utils/types/circuit_types.hpp"
+#include "aztec3/utils/types/convert.hpp"
+#include "aztec3/utils/types/native_types.hpp"
+
+#include <barretenberg/barretenberg.hpp>
+
+namespace aztec3::circuits::abis {
+
+using aztec3::utils::types::CircuitTypes;
+using aztec3::utils::types::NativeTypes;
+
+template <typename NCT> struct GlobalVariables {
+    using address = typename NCT::address;
+    using fr = typename NCT::fr;
+    using boolean = typename NCT::boolean;
+
+    fr chain_id = 0;
+    fr version = 0;
```

Comment:
can we rename `version` to `aztecVersion` and `chainId` to `baseChainID` to make it more clear what these are. Otherwise, especially for noir devs this will be super super confusing

Reply:
Technically they are both chainIds, so could also be `baseChainId` and `rollupChainId` 🤷. Version mainly chosen just to be different to `chainId` to make it clear it was not the same thing.

Reply:
yea I am fine with baseChainId and rollupChainId!